core/src/main/java/org/elasticsearch/client/transport/TransportClient.java

```diff
@@ -112,7 +112,10 @@ private PluginsService newPluginService(final Settings settings) {
                     .put(NettyTransport.PING_SCHEDULE.getKey(), "5s") // enable by default the transport schedule ping interval
                     .put(InternalSettingsPreparer.prepareSettings(settings))
                     .put(NetworkService.NETWORK_SERVER.getKey(), false)
-                    .put(Node.NODE_CLIENT_SETTING.getKey(), true)
+                    //nocommit not too sure if these settings are needed here...
+                    .put(Node.NODE_MASTER_SETTING.getKey(), false)
```

Comment:
why do we need this, we know that we are a transport client, can't we make this explicit somehow?